src/components/Account/SignUp.tsx

```diff
@@ -1,131 +1,318 @@
 import padLock from "../../components/images/account/padlock.png";
 import emailIcon from "../../components/images/account/email.png";
+import { Form, Field } from "react-final-form";
+import FileField from "./FileField";
 
 export default function SignUp() {
-  const [username, setUsername] = useState("");
-  const [password, setPassword] = useState("");
-  const [email, setEmail] = useState("");
-  const [first_name, setFirstName] = useState("");
-  const [last_name, setLastName] = useState("");
+  // const [username, setUsername] = useState("");
+  // const [password, setPassword] = useState("");
+  // const [email, setEmail] = useState("");
+  // const [first_name, setFirstName] = useState("");
+  // const [last_name, setLastName] = useState("");
+  const [avatar, setSelectedAvatar] = useState<any | null>(null);
+  const [registrationStatus, setRegistrationStatus] = useState(true);
   const dispatch = useDispatch();
   const navigate = useNavigate();
-
-  const handleSubmit = (e: any) => {
-    e.preventDefault();
-
-    const newUser = {
-      username,
-      first_name,
-      last_name,
-      email,
-      password,
-    };
-
+  const sleep = (ms:any) => new Promise(resolve => setTimeout(resolve, ms))
+  
+  const onSubmit = (values:any) => {
+    
+    if(!values.bio){
+      values.bio = 'Your biography goes here...'
+    }
+    
+    // await sleep(300)
     fetch(`http://localhost:3000/api/users`, {
       method: "POST",
       headers: {
         Accept: "application/json",
         "Content-Type": "application/json",
       },
-      body: JSON.stringify({ user: newUser }),
+      body: JSON.stringify({user:values}),
     })
       .then((response) => response.json())
       .then((data) => {
-        localStorage.setItem("token", data.token);
-        dispatch(login(data));
+          localStorage.setItem("token", data.token);
+          dispatch(login(data.user));
+          navigate("/profile");
+      })
+      .catch((error) => {
+        console.log(error);
+        setRegistrationStatus(false);
       });
-      navigate('/profile')
+
+    console.log(registrationStatus);
   };
+
   return (
     <div className="background">
-      <div className="signup-container" style={{ padding: "130px" }}>
+      <div className="signup-container" >
+        <Form
+          onSubmit={onSubmit}
+          validate={(values) => {
+            const errors: any = {};
+            if (!values.first_name) {
+              errors.first_name = "First name required";
+            }
+            if (!values.last_name) {
+              errors.last_name = "Last name required";
+            }
+            if (!values.email) {
+              errors.email = "Email required";
+            }
+            if (!values.password) {
+              errors.password = "Password required";
+            }
+            if (!values.username) {
+              errors.username = "Username required";
+            }
```

Comment:
We should probably extract this into an object and export this instead of having it hardcoded here